pageserver/src/tenant.rs

```diff
@@ -752,36 +752,27 @@ impl Tenant {
                     AttachType::Normal
                 };
 
-                let preload_timer = TENANT.preload.start_timer();
-                let preload = match mode {
-                    SpawnMode::Create => {
-                        // Don't count the skipped preload into the histogram of preload durations
-                        preload_timer.stop_and_discard();
+                let preload = match (&mode, &remote_storage) {
+                    (SpawnMode::Create, _) => {
                         None
                     },
-                    SpawnMode::Normal => {
-                        match &remote_storage {
-                            Some(remote_storage) => Some(
-                                match tenant_clone
-                                    .preload(remote_storage, task_mgr::shutdown_token())
-                                    .instrument(
-                                        tracing::info_span!(parent: None, "attach_preload", tenant_id=%tenant_shard_id.tenant_id, shard_id=%tenant_shard_id.shard_slug()),
-                                    )
-                                    .await {
-                                        Ok(p) => {
-                                            preload_timer.observe_duration();
-                                            p
-                                        }
-                                            ,
-                                        Err(e) => {
-                                            make_broken(&tenant_clone, anyhow::anyhow!(e));
-                                                return Ok(());
-                                        }
-                                    },
-                            ),
-                            None => None,
+                    (SpawnMode::Normal, Some(remote_storage)) => {
+                        let _preload_timer = TENANT.preload.start_timer();
+                        let res = tenant_clone
+                            .preload(remote_storage, task_mgr::shutdown_token())
+                            .await;
+                        match res {
+                            Ok(p) => Some(p),
+                            Err(e) => {
+                                make_broken(&tenant_clone, anyhow::anyhow!(e));
+                                return Ok(());
```

Comment:
Now realized it doesn't make sense to return a Result here, it's just to satisfy the taskmgr => we could had noticed the necessity of `make_broken` earlier, or even handle it more generally by bubbling it up by replacing the taskmgr `async`:

```rust
taskmgr::spawn(......, async move {
   let op = async move {
   };
   
   match op {
     Ok(()) => assert!(tenant_clone_clone2.is_active()),
     Err(e) => make_broken(&tenant_clone_clone2, e.into()),
   }
   
   // to satisfy taskmgr
   Ok(())
});
```

Reply:
Well, it's a follow-up anyway because it might be possible to drop 2 indentation at the same time.